proxy/src/main/java/net/md_5/bungee/UserConnection.java

```diff
@@ -771,4 +774,24 @@ public Scoreboard getScoreboard()
     {
         return serverSentScoreboard;
     }
+
+    @Override
+    public CompletableFuture<byte[]> retrieveCookie(String cookie)
+    {
+        return pendingConnection.retrieveCookie( cookie );
+    }
+
+    @Override
+    public void storeCookie(String cookie, byte[] data)
+    {
+        Preconditions.checkState( getPendingConnection().getVersion() >= ProtocolConstants.MINECRAFT_1_20_5, "Cookies are only supported in 1.20.5 and above" );
```

Comment:
Something to discuss - whether we should force a namespace

Reply:
for storeCookie, we don't need to as the client creates a "minecraft:" namespace if no namespace is sent, thats why i added "minecraft:" as default namespace for retrieveCookie so we can run code like this without any issues

```
player.storeCookie("test", "Outfluencer".getBytes());
player.retrieveCookie("test").thenAccept((data) -> {
    player.sendMessage(new String(data));
});
````

Reply:
Yep, I know. But maybe for good API we want to force a namespace